### PR TITLE
Pull clone timeline from peer safekeepers

### PR DESCRIPTION
**Pull request by petuhovskiy**
## Describe your changes

Add HTTP endpoint to initialize safekeeper timeline from peer safekeepers. This is useful for initializing new safekeeper to replace failed safekeeper. Not fully "correct" in all cases, but should work in most.

This code is not suitable for production workloads but can be tested on staging for a start. New endpoint is separated from usual cases and should not affect anything if no one explicitly uses a new endpoint. We can rollback this commit in case of issues.

## Issue ticket number and link

Related to https://github.com/neondatabase/neon/issues/2432. If done properly, it can make #2432 obsolete.

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [X] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by arssher** [comment]
Hm. If this simplifies current procedure in your view (avoiding manual rsync), I don't mind merging it, and right away, as it's better to replace sk ASAP -- it is not invasive (and `display_serialize` > `DisplayFromStr` is good anyway). But we won't need this soon -- it should be replaced by timeline initialization + pulling WAL should going through proper consensus checks.

File: safekeeper/src/pull_timeline.rs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+    let mut statuses = Vec::new();
+    for (i, response) in responses.into_iter().enumerate() {
+        let response = response.context(format!("Failed to get status from {}", http_hosts[i]))?;
+        let status: crate::http::routes::TimelineStatus = response.json().await?;
+        statuses.push((status, i));
+    }
+
+    // Find the most advanced safekeeper
+    // TODO: just using max(commit_lsn) may be wrong, fix it later
```
Comment (arssher) [inline safekeeper/src/pull_timeline.rs:66]: Right, we should compare by (term, flush_lsn) pair.

Comment (petuhovskiy) [reply safekeeper/src/pull_timeline.rs]: Changed comparator to (term, flush_lsn, commit_lsn).`epoch` is probably even better than `term`, but should be ok for now.

**Comment by petuhovskiy** [comment]
Need approve here to deploy to staging @arssher

**Review by arssher — APPROVED** [review]

File: safekeeper/src/pull_timeline.rs (hunk 1 above)
Comment (arssher) [reply safekeeper/src/pull_timeline.rs]: Right, I meant lastLogTerm...

Comment (arssher) [reply safekeeper/src/pull_timeline.rs]: Let's fix it. It is easy, and sorting out consequences would be hard, even if this is staging.

**Pull request merged**